Library/Homebrew/formula_installer.rb

```diff
@@ -492,10 +489,14 @@ def build
     end
 
     ignore_interrupts(:quietly) do # the child will receive the interrupt and marshal it back
+      socket = socket_serv.accept
+      socket.send_io write
       write.close
       data = read.read
       read.close
       Process.wait(pid)
+      socket_serv.close
+      FileUtils.rm_r File.dirname(socket_path)
```

Comment:
The creation of this file should probably be at the very beginning of the method, and then this line in an ensure block at the end.